### DISCUSSION TITLE
SpringBootTest와 WebMvcTest의 차이

### DISCUSSION THREAD

**Discussion by nayonsoso** [2024-woowacourse-study/level-interview#129] [Spring Test]
.

**Comment by J-I-H-O** [comment]
`@SpringBootTest` 는 전체 spring 애플리케이션 컨텍스트를 로드합니다.
모든 빈을 로드하기 때문에 테스트 실행 시간이 비교적 오래 걸리지만, 실제 환경과 가장 유사하게 테스트할 수 있습니다.
전체 애플리케이션의 통합 테스트를 수행할 때 사용할  수 있습니다. 

`@WebMvcTest` 는 웹 계층과 관련된 컴포넌트(Controller, ControllerAdvice, Filter 등)만 로드합니다. 이러한 이유로 `@SpringBootTest` 를 사용할 때에 비해 테스트 실행 시간이 빠릅니다. 
주로 단위 테스트를 수행할 때 사용하며, 테스트 대상 컨트롤러의 의존성은 직접 Mock 객체를 주입하여 테스트 해야합니다.

**Comment by le2sky** [comment]
`@WebMvcTest`는 스프링 MVC를 위한 테스트에서 사용되며, 컨트롤러가 예상대로 작동되는지 검증하기 위해 사용됩니다. 
Web과 관련된 빈들만 스캔하여 가볍게 테스트할 수 있습니다. (`@Controller`, `@ControllerAdvice`...)
반면, `@SpringBootTest`는 모든 스프링 빈을 등록하여 테스트를 진행합니다.
운영환경과 가장 유사한 테스트가 가능하지만, 모든 빈을 로드하기 때문에 시간이 오래 걸릴 수 있습니다.
